cpp/tensorrt_llm/thop/moeOp.cpp

```diff
@@ -405,136 +236,102 @@ class FusedMoeRunner : public torch::CustomClassHolder
         return output;
     }
 
-private:
-    struct WorkspaceInfo
+    int64_t getTacticNum()
     {
-        void* workspace{};
-        void* src_to_dest_map{};
-    };
-
-    std::mutex mMutex;
-    std::shared_ptr<kernels::CutlassMoeFCRunnerInterface> mKernelRunner;
-    std::shared_ptr<kernels::GemmProfilerBackend> mProfiler;
-    std::shared_ptr<MNKProfileMap> mMNKProfileMap;
-    int64_t mMinDimM;
-    int64_t mMaxDimM;
-    c10::ScalarType mActivationDtype;
-    c10::ScalarType mWeightDtype;
-    c10::ScalarType mOutputDtype;
-    // number of elements packed into the inner dimension of a matrix
-    // e.g. 16 nvfp4 elements are packed into a single int64 element
-    int64_t mInnerDimMultiplier;
-
-    bool mUseFp8BlockScaling = false;
-
-    using Profile = tensorrt_llm::cutlass_extensions::CutlassGemmConfig;
-    std::vector<Profile> mAllProfiles;
+        std::lock_guard<std::mutex> lock(mMutex);
+        return mAllProfiles.size();
+    }
 
-    void runProfileGemmIdx(int64_t const hidden_size, int64_t const inter_size, int const num_experts,
-        int const experts_per_token, int const tp_size, int const tp_rank, int const ep_size, int const ep_rank,
-        std::vector<int64_t> const& num_token_buckets, profiler_backend::GemmToProfile const gemm_idx,
-        cudaStream_t stream)
+    void runGemmProfile(torch::Tensor const& input, torch::Tensor const& fc2_expert_weights, int64_t const top_k,
+        int64_t const tp_size, int64_t const tp_rank, int64_t const ep_size, int64_t const ep_rank,
+        int64_t const gemm_idx, int64_t const profile_id, bool const do_preparation)
     {
-        auto gemm_id_moe = GemmIDMoe{gemm_idx, hidden_size, inter_size, num_experts, experts_per_token};
+        std::lock_guard<std::mutex> lock(mMutex);
 
-        if (mMNKProfileMap->existsMProfileMap(gemm_id_moe))
+        if (mUseFp8BlockScaling)
         {
             return;
         }
 
-        mMNKProfileMap->createMProfileMap(gemm_id_moe);
+        int64_t num_rows = input.sizes()[0];
```

Comment:
Nit.

If we use "const", lets use them consistently, I think 'num_rows', "hidden_size", "inter_size" are all "int64_t const"?